### PR TITLE
rebalance rock ressource generation

### PR DESCRIPTION
**Pull request by antists18**
### All Submissions:

- [x] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?

### Tested Changes:

1. [x] Have you lint your code (lua lint) locally prior to submission?

### Changes to Core Features:

- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

**Comment by antists18** [comment]
rebalance the yield, size and rate of ore patches

File: maps/wasteland/rocks_yield_ore_veins.lua [hunk 1]
```diff
@@ -2,21 +2,19 @@ local math_random = math.random
 local math_floor = math.floor
 local table_insert = table.insert
 local table_shuffle = table.shuffle_table
-
 local ScenarioTable = require 'maps.wasteland.table'
-
 local valid_entities = {
```
Comment (blubFisch) [inline maps/wasteland/rocks_yield_ore_veins.lua]: Extra deletions

File: maps/wasteland/rocks_yield_ore_veins.lua [hunk 2]
```diff
@@ -136,7 +135,14 @@ local function ore_vein(event)
...
     local position = event.entity.position
-    local ore_entities = {{ore = {name = ore, position = {x = position.x, y = position.y}}, amount = get_amount()}}
+    local ore_entities = {
+        {
+            ore = {
+                name = ore, position = {
+                    x = position.x, y = position.y}
+            }, amount = get_amount()
+        }
+    }
```
Comment (blubFisch) [inline maps/wasteland/rocks_yield_ore_veins.lua]: Please avoid reformatting

File: maps/wasteland/rocks_yield_ore_veins.lua [hunk 3]
```diff
@@ -178,22 +184,23 @@ local function ore_vein(event)
 end
 
 local function on_player_mined_entity(event)
-    local rocks_yield_ore_veins = ScenarioTable.get('rocks_yield_ore_veins')
+    local this = ScenarioTable.get_table()
+    local rocks_yield_ore_veins = this.rocks_yield_ore_veins
     if not rocks_yield_ore_veins then
```
Comment (blubFisch) [inline maps/wasteland/rocks_yield_ore_veins.lua]: Simplify please

File: maps/wasteland/wreckage_yields_scrap.lua [hunk 4]
```diff
@@ -134,8 +134,8 @@ local function on_player_mined_entity(event)
 
     local scrap = scrap_raffle[math.random(1, size_of_scrap_raffle)]
 
-    local scrap_amount_modifier = 3
-    local amount_bonus = scrap_amount_modifier * (game.forces.enemy.evolution_factor * 2) + (game.forces.player.mining_drill_productivity_bonus * 2)
+    local scrap_amount_modifier = 6
+    local amount_bonus = scrap_amount_modifier + (game.forces.player.mining_drill_productivity_bonus * 2)
```
Comment (blubFisch) [inline maps/wasteland/wreckage_yields_scrap.lua]: Untentional

**Review by blubFisch — CHANGES_REQUESTED** [review]

File: maps/wasteland/rocks_yield_ore_veins.lua (hunk 2 above)
Comment (antists18) [reply maps/wasteland/rocks_yield_ore_veins.lua]: done

File: maps/wasteland/rocks_yield_ore_veins.lua (hunk 3 above)
Comment (antists18) [reply maps/wasteland/rocks_yield_ore_veins.lua]: done

File: maps/wasteland/wreckage_yields_scrap.lua (hunk 4 above)
Comment (antists18) [reply maps/wasteland/wreckage_yields_scrap.lua]: sry bro

**Comment by antists18** [comment]
done

File: maps/wasteland/rocks_yield_ore_veins.lua [hunk 5]
```diff
@@ -97,6 +96,7 @@ end
 local function ore_vein(event)
     local this = ScenarioTable.get_table()
     local surface = event.entity.surface
+    --# means lenght of the string or table
```
Comment (blubFisch) [inline maps/wasteland/rocks_yield_ore_veins.lua]: Please no language explanations in code

**Review by blubFisch — CHANGES_REQUESTED** [review]

File: maps/wasteland/rocks_yield_ore_veins.lua (hunk 5 above)
Comment (antists18) [reply maps/wasteland/rocks_yield_ore_veins.lua]: done and ty

**Pull request merged**